### PR TITLE
WIP: Add override for metric type.

### PR DESCRIPTION
**Pull request by SuperQ**
# TODO
* [ ] Validate config for allowed types.
* [ ] Add tests.

### CODE REVIEW

File: generator/README.md [hunk 1]
```diff
@@ -98,6 +98,7 @@ modules:
                value: '1'
              - regex: '.*'
                value: '0'
+         type: [counter, gauge, DisplayString, ...] # Override the metric type.
```
Comment (brian-brazil) [inline generator/README.md]: You should link over to the FORMAT doc as to what these are

File: generator/tree.go [hunk 2]
```diff
@@ -214,6 +214,9 @@ func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*
 					if !ok {
 						log.Fatalf("Unknown index type %s for %s", indexNode.Type, lookup.NewIndex)
 					}
+					if cfg.Overrides[sanitizeLabelName(indexNode.Label)].Type != "" {
```
Comment (brian-brazil) [inline generator/tree.go]: This is not the only place types are used, you need to produce a copy of the tree with this changed.

**Comment by SuperQ** [comment]
Yes, I know this needs a bunch more work.

**Comment by brian-brazil** [comment]
Obsoleted by #294

**Pull request closed without merging**